compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -938,6 +940,36 @@ impl<'a> Parser<'a> {
             && self.look_ahead(1, |tok| tok.kind == TokenKind::OpenDelim(Delimiter::Parenthesis))
             && let Some(path) = self.recover_path_from_fn()
         {
+            path
+        } else if !self.token.is_path_start() && self.token.can_begin_type() {
+            // Instead of finding a path (a trait), we found a type.
+            let mut err = self.struct_span_err(self.token.span, "expected a trait, found type");
+
+            // If we can recover, try to extract a path from the type. Note
+            // that we do not use the try operator when parsing the type because
+            // if it fails then we get a parser error which we don't want (we're trying
+            // to recover from errors, not make more).
+            let path = if self.may_recover()
+                && let Ok(ty) = self.parse_ty_no_plus()
+                && matches!(ty.kind, TyKind::Ptr(..) | TyKind::Ref(..))
+                && let TyKind::Path(_, path) = &ty.peel_refs().kind {
+                // Just get the indirection part of the type.
+                let span = ty.span.until(path.span);
+
+                err.span_suggestion(
```

Comment:
```suggestion
                err.span_suggestion_verbose(
```
and then re`--bless`.

Reply:
Done, thanks!